lib/internal/http2/util.js

```diff
@@ -221,60 +221,65 @@ const IDX_OPTIONS_MAX_SETTINGS = 9;
+      index: IDX_OPTIONS_MAX_OUTSTANDING_PINGS,
+    },
+    maxOutstandingSettings: {
+      flag: 1 << IDX_OPTIONS_MAX_OUTSTANDING_SETTINGS,
+      index: IDX_OPTIONS_MAX_OUTSTANDING_SETTINGS,
+      min: 1,
+    },
+    maxSessionMemory: {
+      flag: 1 << IDX_OPTIONS_MAX_SESSION_MEMORY,
+      index: IDX_OPTIONS_MAX_SESSION_MEMORY,
+      min: 1,
+    },
+    maxSettings: {
+      flag: 1 << IDX_OPTIONS_MAX_SETTINGS,
+      index: IDX_OPTIONS_MAX_SETTINGS,
+      min: 1,
+    },
+  };
+
   let flags = 0;
-  if (typeof options.maxDeflateDynamicTableSize === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_DEFLATE_DYNAMIC_TABLE_SIZE);
-    optionsBuffer[IDX_OPTIONS_MAX_DEFLATE_DYNAMIC_TABLE_SIZE] =
-      options.maxDeflateDynamicTableSize;
-  }
-  if (typeof options.maxReservedRemoteStreams === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_RESERVED_REMOTE_STREAMS);
-    optionsBuffer[IDX_OPTIONS_MAX_RESERVED_REMOTE_STREAMS] =
-      options.maxReservedRemoteStreams;
-  }
-  if (typeof options.maxSendHeaderBlockLength === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_SEND_HEADER_BLOCK_LENGTH);
-    optionsBuffer[IDX_OPTIONS_MAX_SEND_HEADER_BLOCK_LENGTH] =
-      options.maxSendHeaderBlockLength;
-  }
-  if (typeof options.peerMaxConcurrentStreams === 'number') {
-    flags |= (1 << IDX_OPTIONS_PEER_MAX_CONCURRENT_STREAMS);
-    optionsBuffer[IDX_OPTIONS_PEER_MAX_CONCURRENT_STREAMS] =
-      options.peerMaxConcurrentStreams;
-  }
-  if (typeof options.paddingStrategy === 'number') {
-    flags |= (1 << IDX_OPTIONS_PADDING_STRATEGY);
-    optionsBuffer[IDX_OPTIONS_PADDING_STRATEGY] =
-      options.paddingStrategy;
-  }
-  if (typeof options.maxHeaderListPairs === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_HEADER_LIST_PAIRS);
-    optionsBuffer[IDX_OPTIONS_MAX_HEADER_LIST_PAIRS] =
-      options.maxHeaderListPairs;
-  }
-  if (typeof options.maxOutstandingPings === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_OUTSTANDING_PINGS);
-    optionsBuffer[IDX_OPTIONS_MAX_OUTSTANDING_PINGS] =
-      options.maxOutstandingPings;
-  }
-  if (typeof options.maxOutstandingSettings === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_OUTSTANDING_SETTINGS);
-    optionsBuffer[IDX_OPTIONS_MAX_OUTSTANDING_SETTINGS] =
-      MathMax(1, options.maxOutstandingSettings);
-  }
-  if (typeof options.maxSessionMemory === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_SESSION_MEMORY);
-    optionsBuffer[IDX_OPTIONS_MAX_SESSION_MEMORY] =
-      MathMax(1, options.maxSessionMemory);
-  }
-  if (typeof options.maxSettings === 'number') {
-    flags |= (1 << IDX_OPTIONS_MAX_SETTINGS);
-    optionsBuffer[IDX_OPTIONS_MAX_SETTINGS] =
-      MathMax(1, options.maxSettings);
+  for (const optionKey in optionMap) {
+    const option = optionMap[optionKey];
+    const value = options[optionKey];
+    if (typeof value === 'number') {
+      flags |= option.flag;
+      optionsBuffer[option.index] = option.min ? Math.max(1, value) : value;
```

Comment:
this feels a lot less readable than the previous version - what about this is optimized?